astropy/utils/xml/tests/test_iterparse.py

```diff
@@ -0,0 +1,130 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+# LOCAL
+from ....utils.xml.iterparser import _fast_iterparse
+
+# SYSTEM
+try:
+    # Python 2.x
```

Comment:
Astropy comes bundled with `six`, so you can replace the whole `try ... except ...` block here with
```python
from ....extern import six
```
And then where needed, just use `six.BytesIO`.